### DISCUSSION TITLE
Additional button to "Execute Statement" without having to highlight it.

### DISCUSSION THREAD

**Discussion by robosong** [getredash/redash#6433] [Ideas]
I know it's currently possible to highlight part of my SQL script and click "Execute Selected" but highlighting is tedious if I have multiple SQL statetements in a script and just want to select and execute one of them. Redash should be able to figure out where the statement starts and ends so I shouldn't have to highlight it.

I suggest copying DBeaver (another greate DB management tool) which always shows two execute buttons: "Execute Statement" and "Execute Script". If you click "Execute Statement" it executes the current SQL statement where the cursor is located (no highlighting required). This allows MUCH faster selection and execution of individual SQL statements within a script. When code is highlighted it is still useful to change the button "Execute Statement" to "Execute Selected".

A simple way to determine beginning and end of a statement may be to require using a semicolon at the end of every SQL statement and use that as a statement delimiter within a script. DBeaver is a bit smarter than that and does a good job of detecting statement start/end even without semicolons.

**Comment by guidopetri** [comment]
@robosong thanks again for pointing this out - this is a great idea, I also love how DBeaver does it. I opened an issue in #6435 and will close this discussion accordingly.
